src/components/AddItemForm/AddItemForm.js

```diff
@@ -0,0 +1,115 @@
+import { useState } from 'react';
+import { db } from '../../lib/firebase.js';
+
+const AddItemForm = ({ listId }) => {
+  const defaultFormValues = {
+    itemName: '',
+    purchaseInterval: '7',
+  };
+
+  const [formValues, setFormValues] = useState(defaultFormValues);
+
+  // generic function updates formValues state for any of the below form inputs
+  const handleChange = (event) => {
+    const inputName = event.target.name;
+    setFormValues({ ...formValues, [inputName]: event.target.value });
+  };
+
+  // on form submit, add user input and other default values to database
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    const newItem = formValues;
+    newItem.purchaseInterval = parseInt(newItem.purchaseInterval); // convert purchaseInterval to integer
+    newItem.lastPurchaseDate = null;
+    newItem.numberOfPurchases = 0;
+
+    try {
+      await db.collection(`lists/${listId}/items`).add(formValues); // add item to Firestore database
+      setFormValues(defaultFormValues); // after saving to database, reset form values to defaults
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  return (
+    <form name="addItemForm" onSubmit={handleSubmit} className="add-item-form">
+      <label
+        className="add-item-form__label add-item-form__label_type_text label"
+        htmlFor="itemName"
+      >
+        Item name:
+      </label>
+      <input
+        className="add-item-form__text-field text-field"
+        type="text"
+        id="itemName"
+        name="itemName"
+        aria-describedby="itemNameHint"
+        value={formValues.itemName}
+        onChange={handleChange}
+        maxLength="100"
+        required
+      />
+      <span id="itemNameHint"></span>
```

Comment:
I agree with @Cjstoney's comment above regarding creating an issue for changes that you would like to implement in the future as opposed to leaving partial elements in the code. Particularly if they are not part of the AC.  You can mock out what you would like the code to be and include that in the issue description so that you have the needed reference to follow your original thought on it.

While the information you left here explaining your thoughts is helpful, on a bigger team there will be devs who will come upon this code and not have been involved with the original code review. In those cases they will not have the context to understand the intention behind this span.

Also, it is not uncommon in a project to meet the AC and continue moving forward in which case it might be weeks/months/years before this particular refinement is addressed (I'm currently discussing with my team a comment in our code base that was left to note a portion of code that should 'eventually' be refactored out and the comment is 3+ years old 😁 )

Reply:
Thanks for that helpful insight! I definitely agree that makes sense. Though TBH I didn't realize we'd left that span in there until I reviewed my own PR the next day. 😅

> I did want to point out that it looks like something happened when you updated as the `package-lock.json` added a large amount of code changes. Upon inspecting it locally it looks like many, if not all, packages were duplicated as `node-modules/.....`

Any idea what might have happened there? @connietran-dev mentioned something about a change in one of the packages files on her computer when we started, I think after running npm update (is that right Connie?). I assumed it was just a minor package update and wasn't paying a lot of attention to those files, whoops. Seems to be fixed on my end after running npm update as you suggested, and I pushed the fix.

Reply:
>Any idea what might have happened there?

I didn't when I saw it last night while reviewing, but I dug in a bit today and it is because @connietran-dev is using a newer version of `npm`.  The [updated v7](https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json) made significant changes to the lockfile for performance reasons but needed to stay backward compatible so they kept all the previous settings as well.  This explains why the information appeared to be in there twice.

I found this [blog/video](https://nitayneeman.com/posts/catching-up-with-package-lockfile-changes-in-npm-v7/) to be helpful in breaking down the changes made in npm v7.

We can discuss this at the sync tomorrow and decide on which version we should all use so that we don't end up with this file changing every week depending on who ran an update.